app/games/sudoku.tsx

```diff
@@ -0,0 +1,221 @@
+import { Alert, Dimensions, TextInput, View } from "react-native"; // eslint-disable-line senso-import-sources
+import { Text, Button } from "react-native-paper";
+
+import { actions } from "@/common/actions";
+import { useMutation } from "@/common/api";
+import { AppRoutes } from "@/common/constants";
+import { useI18n } from "@/common/i18n";
+import { calculateScore } from "@/common/score";
+import { sty } from "@/common/styles";
+import { Header } from "@/components";
+import {
+  checkBoard,
+  emptyCellCount,
+  generateSolvedSudoku,
+  generateSudoku,
+} from "@/logic/sudoku";
+
+const screen = {
+  width: Dimensions.get("window").width,
+  height: Dimensions.get("window").height,
+};
+
+const calculateBorder = (rowIndex: number, colIndex: number) => {
+  let borderStyle = {};
+
+  if (rowIndex >= 0 && rowIndex <= 7) {
+    borderStyle = {
+      ...borderStyle,
+      borderBottomWidth: rowIndex === 2 || rowIndex === 5 ? 2 : 1,
+    };
+  }
+
+  if (colIndex >= 0 && colIndex <= 7) {
+    borderStyle = {
+      ...borderStyle,
+      borderRightWidth: colIndex === 2 || colIndex === 5 ? 3 : 1,
+    };
+  }
+
+  return borderStyle;
+};
+
+const SudokuGame = () => {
+  const { t } = useI18n();
+  const styles = useStyles();
+  const timer = useRef<NodeJS.Timeout | null>(null);
+  const [seconds, setSeconds] = useState(0);
+  const [checks, setChecks] = useState(0);
+  const [gameStarted, setGameStarted] = useState(false);
+  const [solvedBoard] = useState<number[][]>(() => generateSolvedSudoku());
+  const [[cellsToFill, board], setBoard] = useState<
+    [cellsToFill: { row: number; col: number }[], board: number[][]]
+  >(() => generateSudoku(emptyCellCount, solvedBoard));
+  const postNewScore = useMutation("post", "/api/v1/games/{gameName}/score");
+
+  // Function to handle digit input
+  const handleDigitInput = (row: number, col: number, value: number) => {
+    setBoard((prevBoard) => {
+      const newBoard = [...prevBoard[1]];
+      newBoard[row]![col] = value < 1 || value > 9 ? 0 : value;
+      return [prevBoard[0], newBoard];
+    });
+  };
+
+  const showResult = () => {
+    const result = checkBoard(board);
+    setChecks((prevChecks) => prevChecks + 1);
+    if (result) {
+      setGameStarted(false);
+      void postNewScore({
+        params: {
+          path: {
+            gameName: "sudoku",
+          },
+        },
+        body: {
+          score: calculateScore(checks, seconds),
+        },
+      });
+    }
+    Alert.alert(
+      result
+        ? t("games.sudoku.alertTitlePositive")
+        : t("games.sudoku.alertTitleNegative"),
+      result ? t("games.sudoku.alertDescription", { time: seconds }) : "",
+      [
+        {
+          text: result
+            ? t("games.sudoku.backToMenu")
+            : t("games.sudoku.continue"),
+          onPress: () => (result ? router.replace(AppRoutes.Dashboard) : null),
+        },
+      ],
+    );
+  };
+
+  useEffect(() => {
+    setGameStarted(true);
+  }, []);
```

Comment:
**Q:** Why this instead of just initializing `gameStarted` to `true` instead of `false`? I was wondering the same thing when looking through Wordle's code recently.

Reply:
You're right, will change that here and in wordle game 🤚🏼 good catch